### PR TITLE
Fix 2159  The user is not able to add audio and video from Google Drive

### LINKED ISSUE #2159: The user is not able to add audio and video from Google Drive
#### Software and hardware versions 
Collect master
Android 5.1, 6.0, 7.0, 8.0

#### Problem description
The user is not able to add audio and video from Google Drive

#### Steps to reproduce the problem
1. The user tries to add an audio file from Google Drive
2. The user tries to add video file from Google Drive
3. The user sees message and file is not attached

| | |
|---|---|
|screenshot_20180427-101524|screenshot_20180427-101752|

#### Expected behavior
1. User is able to add audio and video file from Google Drive

### PR DESCRIPTION
**Pull request by ShenJack**
FIX #2159 The user is not able to add audio and video from Google Drive

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/utilities/MediaUtils.java [hunk 1]
```diff
@@ -619,6 +619,7 @@ public static boolean isGooglePhotosUri(Uri uri) {
      */
     public static boolean isGoogleDriveDocument(Uri uri) {
         return "com.google.android.apps.docs.storage".equals(uri.getAuthority())
+                || "com.google.android.apps.docs.storage.legacy".equals(uri.getAuthority())
                 || uri.getAuthority().startsWith("com.google.android.apps.photos.content");
```
Comment (ShenJack) [inline collect_app/src/main/java/org/odk/collect/android/utilities/MediaUtils.java]: I found that there are two "google drive" tab in my file manager , and their returned  Uri's authorities are different ,so i add
 "com.google.android.apps.docs.storage.legacy"  for supporting legacy Google Drive File.

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java [hunk 2]
```diff
@@ -838,6 +866,136 @@ public void run() {
...
+    /**
+     * Save a copy of the chosen video in Collect's own path such as
+     * "/storage/emulated/0/odk/instances/{form name}/filename",
+     * and if it's from Google Drive and not cached yet, we'll retrieve it using network.
+     * This may take a long time.
+     *
+     * @param selectedVideo uri of the selected video
+     * @see #getFileExtensionFromUri(Uri)
+     */
+    private void saveChosenVideo(Uri selectedVideo) {
```
Comment (lakshyagupta21) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: `saveChosenVideo` and `saveChosenAudio` doing the same thing, so we can keep that as one single method.

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java [hunk 3]
```diff
@@ -798,8 +799,35 @@ public void run() {
 
 
             case RequestCodes.AUDIO_CHOOSER:
+                /*
```
Comment (lakshyagupta21) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: Both the switch cases calling the same method. 
```
case RequestCodes.AUDIO_CHOOSER:
case RequestCodes.VIDEO_CHOOSER:
                showDialog(SAVING_DIALOG);
                Runnable runnable = new Runnable() {
                    @Override
                    public void run() {
                        saveChosenAudioVideo(intent.getData());
                    }
                };
                new Thread(runnable).start();
                break;
```

Comment (ShenJack) [reply collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: Thanks for reminding me, and if there are any improvement can be done with my code, any suggestions would be appreciated.

**Comment by lakshyagupta21** [comment]
@ShenJack Please fix the conflicts.

**Comment by ShenJack** [comment]
I've opened a new PR for this issue.

**Comment by xsteelej** [comment]
Hi @ShenJack, perhaps this PR should be deleted if https://github.com/opendatakit/collect/pull/2189 now supersedes this one?

**Pull request closed without merging**